Libraries/LibJS/Runtime/RegExpObject.cpp

```diff
@@ -24,13 +24,66 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <AK/Function.h>
 #include <AK/StringBuilder.h>
 #include <LibJS/Heap/Heap.h>
 #include <LibJS/Runtime/GlobalObject.h>
 #include <LibJS/Runtime/PrimitiveString.h>
 #include <LibJS/Runtime/RegExpObject.h>
 #include <LibJS/Runtime/Value.h>
 
+static regex::RegexOptions<ECMAScriptFlags> options_from(const String& flags, JS::VM& vm, JS::GlobalObject& global_object)
+{
+    bool g = false, i = false, m = false, s = false, u = false, y = false;
+    regex::RegexOptions<ECMAScriptFlags> options { (regex::ECMAScriptFlags)regex::AllFlags::Global }; // JS regexps are all 'global' as per our definition, but the "global" flag enables "stateful".
```

Comment:
This is actually pretty weird (as expected of javascript!), all expressions are global by default (as in, they are not anchored), but the 'global' flag makes the instance 'stateful' in that it continues from whatever offset it last left off (if it was successful) and zero otherwise.

Future work (for `exec()` and co) should keep this fact in mind, also the fact that `Regex::match()` will find _all_ matches if `Global` is set; we might want to introduce another internal flag for this, something like `SingleMatch`.